test/utils.go

```diff
@@ -0,0 +1,168 @@
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"os"
+	"os/exec"
+	"strconv"
+	"time"
+
+	"golang.org/x/oauth2/google"
+	cloudresourcemanager "google.golang.org/api/cloudresourcemanager/v1"
+	"google.golang.org/api/option"
+	"k8s.io/klog/v2"
+	boskosclient "sigs.k8s.io/boskos/client"
+	"sigs.k8s.io/boskos/common"
+)
+
+var boskos, _ = boskosclient.NewClient(os.Getenv("JOB_NAME"), "http://boskos", "", "")
+
+// getBoskosProject retries acquiring a boskos project until success or timeout.
+func getBoskosProject(resourceType string) *common.Resource {
+	timer := time.NewTimer(30 * time.Minute)
+	defer timer.Stop()
+	ticker := time.NewTicker(1 * time.Minute)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-timer.C:
+			klog.Fatalf("timed out trying to acquire boskos project")
+		case <-ticker.C:
+			p, err := boskos.Acquire(resourceType, "free", "busy")
+			switch {
+			case err != nil:
+				klog.Warningf("boskos failed to acquire project: %w", err)
+			case p == nil:
+				klog.Warningf("boskos does not have a free %s at the moment", resourceType)
+			default:
+				return p
+			}
+		}
+	}
+}
+
+func setupProwConfig(resourceType string) (string, string) {
+	// Try to get a Boskos project
+	klog.V(4).Info("Running in PROW")
+	klog.V(4).Info("Fetching a Boskos loaned project")
+
+	p := getBoskosProject(resourceType)
+	project := p.Name
+
+	go func(c *boskosclient.Client, _ string) {
+		for range time.Tick(time.Minute * 5) {
+			if err := c.UpdateOne(p.Name, "busy", nil); err != nil {
+				klog.Warningf("[Boskos] Update %s failed with %v", p.Name, err)
+			}
+		}
+	}(boskos, p.Name)
+
+	klog.V(4).Info("Fetching the default compute service account")
+
+	c, err := google.DefaultClient(context.Background(), cloudresourcemanager.CloudPlatformScope)
+	if err != nil {
+		klog.Fatalf("Failed to get Google Default Client: %w", err)
+	}
+
+	cloudresourcemanagerService, err := cloudresourcemanager.NewService(context.Background(), option.WithHTTPClient(c))
+	if err != nil {
+		klog.Fatalf("Failed to create new cloudresourcemanager: %w", err)
+	}
+
+	resp, err := cloudresourcemanagerService.Projects.Get(project).Do()
+	if err != nil {
+		klog.Fatalf("Failed to get project %v from Cloud Resource Manager: %w", project, err)
+	}
+
+	// Default Compute Engine service account
+	// [PROJECT_NUMBER]-compute@developer.gserviceaccount.com
+	serviceAccount := fmt.Sprintf("%v-compute@developer.gserviceaccount.com", resp.ProjectNumber)
+	klog.Infof("Using project %v and service account %v", project, serviceAccount)
+
+	return project, serviceAccount
+}
+
+func setEnvProject(project string) error {
```

Comment:
Is there a way to use the go client instead of using exec?

Reply:
Do you mean using go client to set current project or create google cloud resources?

Reply:
I meant go client to create google cloud resources. For tests, I think it will be easier to have a client than using the cli

Reply:
Yes we can and I started with this. But I found we might need too many clients for different resources when I was implementing some of those tests, like clients for GKE, firewall-rules, subnets, ssl-policies, etc.

I changed all of those to gcloud to keep it simple.  What do you think?

Reply:
why not use k8s-cloud-provider? It's very easy to use:

```

	ctx := context.Background()
	client, err := google.DefaultClient(ctx, compute.ComputeScope)
	if err != nil {
		log.Fatal(err)
	}

	alpha, err := alpha.New(client)
	if err != nil {
		log.Fatal(err)
	}
	beta, err := beta.New(client)
	if err != nil {
		log.Fatal(err)
	}
	ga, err := compute.New(client)
	if err != nil {
		log.Fatal(err)
	}

	svc := &cloud.Service{
		GA:            ga,
		Alpha:         alpha,
		Beta:          beta,
		ProjectRouter: &cloud.SingleProjectRouter{ID: testFlags.project},
		RateLimiter:   &cloud.NopRateLimiter{},
	}
	theCloud = cloud.NewGCE(svc)


	regions, err := theCloud.Regions().List(ctx, nil)
	if err != nil {
		t.Fatalf("Error listing Regions: %v", err)
	}

	const regionName = "us-central1"

	t.Logf("Got %d Regions", len(regions))

	var found bool
	for _, z := range regions {
		if z.Name == regionName {
			found = true
		}
	}
	if !found {
		t.Fatalf("%q was not in the list of Regions", regionName)
	}

	_, err = theCloud.Regions().Get(ctx, meta.GlobalKey(regionName))
	if err != nil {
		t.Fatalf("Get(%q) = _, %v; want _, nil", regionName, err)
	}

```

Reply:
Thanks! Added compute client using k8s-cloud-provider.